### ISSUE TITLE
Exceptoins when creating new object in storage

### ISSUE THREAD

**Issue by medvedev** [quarkiverse/quarkiverse-google-cloud-services#57]
**Issue:** Unable to create object in Storage
**Extension version:** 0.2.0

Both code fragments below work in "JVM" mode, but fail with exceptions in runtime in native mode:
```java
byte[] data = (read from file)
        BlobInfo blobInfo = BlobInfo
                .newBuilder("bucket-name", "id")
//               Adding line below does not change the output
//              .setContentType("image/heif")
                .build();
        storage.create(blobInfo, data); // <- Exception is thrown here
```

Exception:
```
com.google.cloud.storage.StorageException: Unsupported content with type: application/octet-stream
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:227)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.create(HttpStorageRpc.java:308)
	at com.google.cloud.storage.StorageImpl$3.call(StorageImpl.java:213)
	at com.google.cloud.storage.StorageImpl$3.call(StorageImpl.java:210)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.StorageImpl.internalCreate(StorageImpl.java:209)
	at com.google.cloud.storage.StorageImpl.create(StorageImpl.java:171)
```

Tried different method to create an, got different exception.

Code:
```java
        Path filePath = .... (path to the local file to upload)
        BlobInfo blobInfo = BlobInfo
                .build();
        storage.createFrom(blobInfo, filePath); // <- Exception is thrown here
```

Exception:
```
com.google.cloud.storage.StorageException: java.lang.IllegalArgumentException: java.net.MalformedURLException
	at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:77)
	at com.google.cloud.BaseWriteChannel.close(BaseWriteChannel.java:151)
	at com.google.cloud.storage.StorageImpl.createFrom(StorageImpl.java:256)
	at com.google.cloud.storage.StorageImpl.createFrom(StorageImpl.java:237)
	at com.google.cloud.storage.StorageImpl.createFrom(StorageImpl.java:227)
```

**Comment by loicmathieu** [comment]
Hello,
Thanks for reporting this issue.
Can you prepare a small reproducer, you can attach it in the issue or give the link to a github repo.

**Comment by medvedev** [comment]
Will do a bit later today

**Comment by medvedev** [comment]
@loicmathieu 
Created a demo project with all the details needed placed in README.md:
https://github.com/medvedev/quarkus-gcp-storage-write-error

**Comment by medvedev** [comment]
Updated project (link in commend above) to demonstrate both exceptions, added `gsutil rm` to the output for easier cleanup

**Comment by loicmathieu** [comment]
@medvedev thanks, I can reproduce the issue.

I had a quick look and didn't find the issue, will need to digg deeper to try understanding what's going on.

@dzou I know your team starts working on native support for several Google Cloud Services, do you have this kind of issues for Storage? I register for reflection all subclasses of `com.google.api.client.json.GenericJson` and all inner classes of ` com.google.api.services.storage.Storage` (and there subclasses, recursively) but this didn't seems to be enought to be able to create new blog on storage (reading existing blob works).

**Comment by dzou** [comment]
Ahh interesting, let me try reproducing this.

We have some basic native image tests for storage; I'm curious if this is a case that we miss as well.

**Comment by dzou** [comment]
Interesting, I was able to fix the error by adding our artifact (containing the reflection configs we've been building up so far): https://github.com/GoogleCloudPlatform/google-cloud-graalvm-support

Basically just did this - added the sonatype snapshots repo:
```
<repositories>
    <repository>
        <id>snapshots-repo</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        <releases><enabled>false</enabled></releases>
        <snapshots><enabled>true</enabled></snapshots>
    </repository>
</repositories>
```

Then our artifact to the deps:
```
<dependency>
    <groupId>com.google.cloud</groupId>
    <artifactId>google-cloud-graalvm-support</artifactId>
    <version>1.0.0-SNAPSHOT</version>
</dependency>
```

And it worked! I think there is some missing reflection configuration that causes this problem to arise.

@loicmathieu - Would you be interested in having your quarkus extension depend on our dependency for the reflection configs? Perhaps with just Pub/Sub and Storage for now as they are the most mature for us. If so, I can write the PR to do this and also add whatever reflection configs we missed here into our repo; maybe I can publish an early version of our dependency so you can avoid relying on snapshot.

We have a CI build to validate our configs; our [Pub/Sub test app](https://github.com/GoogleCloudPlatform/google-cloud-graalvm-support/blob/master/google-cloud-graalvm-samples/graalvm-samples-client-library/pubsub-sample/src/main/java/com/example/PubSubSampleApplication.java#L69) is the most comprehensive where we try to exercise every API call and then compile the app with graalvm, then run to see if it works.

**Comment by loicmathieu** [comment]
@dzou Yes I think using your dependency is the right thing to do.

If you can create a first release so we didn't rely on snapshots it would be cool.

I will then remove reflection config from here and upstream what's missing in the Google native support lib.

A PR is of course very welcome 😀

**Comment by dzou** [comment]
> @dzou Yes I think using your dependency is the right thing to do.
> 
> If you can create a first release so we didn't rely on snapshots it would be cool.
> 
> I will then remove reflection config from here and upstream what's missing in the Google native support lib.
> 
> A PR is of course very welcome

Sure! That's sounds great. I'll get a release out in a few days and send a PR your way shortly!

**Comment by dzou** [comment]
Hi there, we just released version `0.1.0` of `google-cloud-graalvm-support`. It is ready to be used by downstream users.

I sent out a PR to modify the storage extension to depend on our reflection configs here: https://github.com/quarkiverse/quarkiverse-google-cloud-services/pull/58

If you feel this is satisfactory, I can do the same process for Pub/Sub as well!

**Comment by loicmathieu** [comment]
Thanks @dzou I'll test #58 with a reproducer of this one to check that it fixes the issue then we can proceed with the same change for PubSub.

**Comment by loicmathieu** [comment]
This has been fixed by #58 thanks to @dzou and the GCP Java Team :)

It will be included in the next release, no date yet but if it's urgent I can create a release with this fix.
